src/json/from_json.cr

```diff
@@ -66,6 +66,20 @@ def Array.new(pull : JSON::PullParser)
   end
 end
 
+def Set.new(pull : JSON::PullParser)
+  ary = new
+  new(pull) do |element|
+    ary << element
+  end
+  ary
+end
+
+def Set.new(pull : JSON::PullParser)
+  pull.read_array do
+    yield T.new(pull)
+  end
+end
```

Comment:
I'm not sure I agree with the existence of this method. It yields multiple times the same element, which doesn't make much sense for a `Set`. It also doesn't return a `Set`, which violates its name: `new`.